app/screens/post_options/index.js

```diff
@@ -65,12 +77,38 @@ function mapStateToProps(state, ownProps) {
         }
     }
 
+    if (ownProps.channelIsReadOnly) {
+        canFlag = false;
+    }
+
+    if (ownProps.isSystemMessage) {
+        canAddReaction = false;
+        canCopyPermalink = false;
+        canEdit = false;
+        canPin = false;
+    }
+    if (ownProps.hasBeenDeleted) {
+        canDelete = false;
+    }
+
+    if (!ownProps.showAddReaction) {
```

Comment:
can you explain this use case?

Reply:
I was trying to move the logic from `71 if (showAddReaction && canAddReaction) {` in post_options.js to the connector. I'm not sure how showAddReaction is set! Should I change it back?

Reply:
`showAddReaction` is a prop set in case the post is shown as a search result, is fine to leave it as is